### PR TITLE
Add CRUD endpoints to BookshelvesController

### PR DESCRIPTION
**Pull request by thiagograndesso**

### CODE REVIEW

File: src/VintageBookshelf.Data/Repository/BookshelfRepository.cs [hunk 1]
```diff
@@ -7,5 +10,20 @@ namespace VintageBookshelf.Data.Repository
     public sealed class BookshelfRepository : Repository<Bookshelf>, IBookshelfRepository
     {
         public BookshelfRepository(VintageBookshelfContext db) : base(db) { }
+        public async Task<Bookshelf> GetBookshelfWithProducts(long id)
```
Comment (dieguezthiago) [inline src/VintageBookshelf.Data/Repository/BookshelfRepository.cs:13]: Why don't you create a private **Private Task IQueriable<Bookshelf> GetBookshelfWithProducts() => ...** ? Then you can call it inside the other methods.

File: src/VintageBookshelf.Api/Controllers/BookshelvesController.cs [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+            await _bookshelfService.Update(_mapper.Map<Bookshelf>(bookshelfDto));
+            return CustomResponse(bookshelfDto);
+        }
+        
+        [HttpDelete]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        public async Task<ActionResult<BookshelfDto>> Remove(long id)
+        {
+            var bookshelf = _mapper.Map<BookshelfDto>(await _bookshelfRepository.GetById(id));
```
Comment (dieguezthiago) [inline src/VintageBookshelf.Api/Controllers/BookshelvesController.cs]: Do you really need to map the object?

File: src/VintageBookshelf.Api/Controllers/BookshelvesController.cs [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+            return CustomResponse(bookshelfDto);
+        }
+        
+        [HttpPut("{id:long}")]
+        [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(BookshelfDto))]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        public async Task<ActionResult<BookshelfDto>> Update(long id, [FromBody] BookshelfDto bookshelfDto)
+        {
+            if (id != bookshelfDto.Id)
+            {
```
Comment (dieguezthiago) [inline src/VintageBookshelf.Api/Controllers/BookshelvesController.cs]: Considering the Id in the route, I would just remove the Id from BooksheldDto.

**Review by dieguezthiago — APPROVED** [review]

File: src/VintageBookshelf.Api/Controllers/BookshelvesController.cs (hunk 2 above)
Comment (thiagograndesso) [reply src/VintageBookshelf.Api/Controllers/BookshelvesController.cs]: Nice catch, it's really not necessary to map domain to dto model

File: src/VintageBookshelf.Api/Controllers/BookshelvesController.cs (hunk 3 above)
Comment (thiagograndesso) [reply src/VintageBookshelf.Api/Controllers/BookshelvesController.cs]: Yes, it seems to just be redundant. Another nice catch!

**Pull request merged**